### ISSUE TITLE
Deployment failed but mup deploy returned code 0

### ISSUE THREAD

**Issue by artgo** [zodern/meteor-up#673]
In case if anything during deployment fails `mup deploy` must return non-zero error code or it leads to a lot of confusion. Currently, it returns 0 if deployment fails due to incorrectly formed settings.json

```
> mup deploy
...
[10.10.10.10] - Pushing the Startup Script: SUCCESS
(node:23706) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): SyntaxError: /home/ubuntu/mupx_prod/settings.json: Unexpected token } in JSON at position 516
(node:23706) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

>
> echo $?
0
```

**Comment by zodern** [comment]
What mup version are you using (`mup --version`)?

This is similar to #320 and #137.

**Comment by artgo** [comment]
```
$ mup --version

  error: unknown option `--version'
```
But we explicitly install 1.0.4 because we use Meteor 1.3.5.1

And yes, I believe this ticket is a special case for https://github.com/zodern/meteor-up/issues/320 which might or might not be fixed by https://github.com/zodern/meteor-up/issues/320

**Comment by zodern** [comment]
I tried this with the 1.3 beta, and the error code is now 1 when it is unable to load or parse the settings.json.  Though it still is 0 when a task list fails, which will be fixed by #320.
